### PR TITLE
HPCC-15619 Enhance WsESDLConfig getESDLBinding method

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review (I'm specially interested in hearing your thoughts on the ECM interface change)

File: esp/scm/ws_esdlconfig.ecm [hunk 1]
```diff
@@ -101,6 +101,45 @@ ESPrequest GetESDLBindingRequest
...
+ESPStruct Attribute
+{
+	String Name;
+	String Value;
+};
+
+ESPStruct MethodConfig
+{
+	String Name;
+	ESParray<ESPStruct Attribute, Attribute> Attributes;
```
Comment (wangkx) [inline esp/scm/ws_esdlconfig.ecm]: If you like, you may use:
ESParray<ESPstruct NamedValue, Attribute> Attributes;

The ESPstruct NamedValue is defined in common.ecm

File: esp/scm/ws_esdlconfig.ecm [hunk 2]
```diff
@@ -109,7 +148,8 @@ ESPresponse [exceptions_inline] GetESDLBindingResponse
     string EspProcName;   //Name of ESP Process
     string BindingName;
     string EspPort;
-    string ConfigXML;
+    [max_ver("1.1")] String ConfigXML;
```
Comment (wangkx) [inline esp/scm/ws_esdlconfig.ecm]: max_ver or depr_var?

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 3]
```diff
@@ -917,55 +917,90 @@ bool CWsESDLConfigEx::onConfigureESDLBindingMethod(IEspContext &context, IEspCon
...
+        return conn->getRoot();
     }
-    conn->close(false);
-    globalLock->close(false);
-    return 0;
+}
+
+int CWsESDLConfigEx::getBindingXML(const char * espProcName, const char * espBindingName, StringBuffer & bindingXml, StringBuffer & msg)
+{
+    IPropertyTree * esdlBinding = getBindingTree(espProcName, espBindingName, msg);
```
Comment (wangkx) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: use Owned?

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 4]
```diff
@@ -1039,8 +1074,77 @@ bool CWsESDLConfigEx::onGetESDLBinding(IEspContext &context, IEspGetESDLBindingR
...
+					{
+						fetchESDLDefinitionFromDaliById(defid.toLowerCase(), definition);
+					}
+					catch (...)
+					{
+						msg.appendf("\nUnexpected error while attempting to fetch ESDL Definition %s", defid.toLowerCase().str());
+					}
+
+					resp.updateESDLBinding().updateDefinition().setInterface(definition.str());
+					msg.append("\nFetched ESDL Biding definition.");
```
Comment (wangkx) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: Should those 2 lines be inside the try{}?

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 5]
```diff
@@ -1039,8 +1074,77 @@ bool CWsESDLConfigEx::onGetESDLBinding(IEspContext &context, IEspGetESDLBindingR
...
+				}
```
Comment (wangkx) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: bad indent

**Comment by wangkx** [comment]
@rpastrana back to you

File: esp/scm/ws_esdlconfig.ecm (hunk 2 above)
Comment (rpastrana) [inline esp/scm/ws_esdlconfig.ecm]: depr_var sounds good too, what's the difference between the two?

**Comment by rpastrana** [comment]
@wangkx please review latest changes

**Comment by wangkx** [comment]
@rpastrana looks good.

**Pull request merged**